hoplite-core/src/test/kotlin/com/sksamuel/hoplite/SystemPropertiesPropertySourceTest.kt

```diff
@@ -0,0 +1,59 @@
+package com.sksamuel.hoplite
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.maps.shouldContainExactly
+import io.kotest.matchers.shouldBe
+import java.util.Properties
+class SystemPropertiesPropertySourceTest : FunSpec() {
+  init {
+    test("reads config from string") {
+      data class TestConfig(val a: String, val b: Int)
+
+      val config = ConfigLoader {
+        withDefaultSources(false)
+        addPropertySource(
+          SystemPropertiesPropertySource(
+            systemPropertiesMap = {
+              mapOf("config.override.a" to "Overridden by system prop")
+            },
```

Comment:
The goal is to allow consumers of `hoplite` to write tests like this.

Reply:
awesome